Missing docstrings

## Description
- write any missing docstrings

## User Story
As an engineer it is important to hand off work which has proper documentation and descriptions.

### Missing Docstrings
- Found after `pylint * .py`

************* Module extract.steam.test_extract_steam
extract/steam/test_extract_steam.py:3:0: E0401: Unable to import 'steam_extract' (import-error)
************* Module extract.steam.conftest
extract/steam/conftest.py:8:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/steam/conftest.py:16:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/steam/conftest.py:23:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/steam/conftest.py:30:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/steam/conftest.py:37:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/steam/conftest.py:44:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/steam/conftest.py:52:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/steam/conftest.py:62:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/steam/conftest.py:70:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/steam/conftest.py:92:0: C0116: Missing function or method docstring (missing-function-docstring)

************* Module extract.gog.conftest
extract/gog/conftest.py:1:0: C0114: Missing module docstring (missing-module-docstring)
extract/gog/conftest.py:7:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/gog/conftest.py:16:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/gog/conftest.py:24:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/gog/conftest.py:31:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/gog/conftest.py:38:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/gog/conftest.py:45:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/gog/conftest.py:51:0: C0116: Missing function or method docstring (missing-function-docstring)

************* Module extract.epic.conftest
extract/epic/conftest.py:14:0: C0301: Line too long (103/100) (line-too-long)
extract/epic/conftest.py:1:0: C0114: Missing module docstring (missing-module-docstring)
extract/epic/conftest.py:5:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/epic/conftest.py:66:0: C0116: Missing function or method docstring (missing-function-docstring)
extract/epic/conftest.py:118:0: C0116: Missing function or method docstring (missing-function-docstring)

--- a/pipeline/extract/epic/conftest.py
+++ b/pipeline/extract/epic/conftest.py
@@ -66,7 +66,7 @@ def epic_game_post_response() -> dict:
 
 
 @pytest.fixture
-def epic_game_data_game_obj() -> Dict:
+def epic_game_data_game_obj() -> dict:
     '''Returns an example post response.'''
     return {
         "title": "Vengeance of Mr. Peppermint",

--- a/pipeline/extract/epic/conftest.py
+++ b/pipeline/extract/epic/conftest.py
@@ -66,7 +66,7 @@ def epic_game_post_response() -> Dict:
 
 
 @pytest.fixture
-def epic_game_data_game_obj():
+def epic_game_data_game_obj() -> Dict:
     '''Returns an example post response.'''
     return {
         "title": "Vengeance of Mr. Peppermint",

--- a/pipeline/extract/epic/conftest.py
+++ b/pipeline/extract/epic/conftest.py
@@ -1,8 +1,11 @@
+'''A conftest file that defines functions used in test_epic.py.'''
+
 import pytest
 
 
 @pytest.fixture
 def epic_game_post_response():
+    '''Returns an example post response.'''
     return {
         "data": {
             "Catalog": {
@@ -64,6 +67,7 @@ def epic_game_post_response():
 
 @pytest.fixture
 def epic_game_data_game_obj():
+    '''Returns an example post response.'''
     return {
         "title": "Vengeance of Mr. Peppermint",
         "releaseDate": "2024-04-24T16:00:00.000Z",
@@ -116,6 +120,7 @@ def epic_game_data_game_obj():
 
 @pytest.fixture
 def epic_empty_post_request():
+    '''Returns an empty post response.'''
     return {
         "data": {
             "Catalog": {

--- a/pipeline/extract/epic/conftest.py
+++ b/pipeline/extract/epic/conftest.py
@@ -4,7 +4,7 @@ import pytest
 
 
 @pytest.fixture
-def epic_game_post_response():
+def epic_game_post_response() -> Dict:
     '''Returns an example post response.'''
     return {
         "data": {

--- a/pipeline/extract/epic/conftest.py
+++ b/pipeline/extract/epic/conftest.py
@@ -119,7 +119,7 @@ def epic_game_data_game_obj() -> dict:
 
 
 @pytest.fixture
-def epic_empty_post_request() -> Dict:
+def epic_empty_post_request() -> dict:
     '''Returns an empty post response.'''
     return {
         "data": {

--- a/pipeline/extract/epic/conftest.py
+++ b/pipeline/extract/epic/conftest.py
@@ -119,7 +119,7 @@ def epic_game_data_game_obj() -> Dict:
 
 
 @pytest.fixture
-def epic_empty_post_request():
+def epic_empty_post_request() -> Dict:
     '''Returns an empty post response.'''
     return {
         "data": {

--- a/pipeline/extract/epic/conftest.py
+++ b/pipeline/extract/epic/conftest.py
@@ -4,7 +4,7 @@ import pytest
 
 
 @pytest.fixture
-def epic_game_post_response() -> Dict:
+def epic_game_post_response() -> dict:
     '''Returns an example post response.'''
     return {
         "data": {

--- a/pipeline/extract/gog/conftest.py
+++ b/pipeline/extract/gog/conftest.py
@@ -49,7 +49,7 @@ def gog_title() -> BeautifulSoup:
 
 
 @pytest.fixture
-def gog_description():
+def gog_description() -> BeautifulSoup:
     '''Returns an example of a game description.'''
     return BeautifulSoup(
         '''<div class="description" >a very nice really thorough description </div>''', features='html.parser')

--- a/pipeline/extract/gog/conftest.py
+++ b/pipeline/extract/gog/conftest.py
@@ -16,7 +16,7 @@ def gog_fake_links() -> list[BeautifulSoup]:
 
 
 @pytest.fixture
-def gog_fake_links_bad():
+def gog_fake_links_bad() -> list[BeautifulSoup]:
     '''Returns badly formatted hrefs.'''
     return [BeautifulSoup(
         '<a href="/gamesubliesda">publisher1</a>', features='html.parser').find('a'), BeautifulSoup(

--- a/pipeline/extract/gog/conftest.py
+++ b/pipeline/extract/gog/conftest.py
@@ -41,7 +41,7 @@ def gog_tags_bad() -> list[BeautifulSoup]:
 
 
 @pytest.fixture
-def gog_title():
+def gog_title() -> BeautifulSoup:
     '''Returns an example of a game title.'''
     return BeautifulSoup(
         '''<div class="product-tile__title" title='goodtitle'>tag7</div>

--- a/pipeline/extract/gog/conftest.py
+++ b/pipeline/extract/gog/conftest.py
@@ -33,7 +33,7 @@ def gog_tags() -> list[BeautifulSoup]:
 
 
 @pytest.fixture
-def gog_tags_bad():
+def gog_tags_bad() -> list[BeautifulSoup]:
     '''Returns a bad example of a Soup of game tags.'''
     return BeautifulSoup(
         '''<span class="detail">tag7</span>

--- a/pipeline/extract/gog/conftest.py
+++ b/pipeline/extract/gog/conftest.py
@@ -33,7 +33,7 @@ def gog_tags() -> list[BeautifulSoup]:
 
 
 @pytest.fixture
-def gog_tags_bad() -> list[BeautifulSoup]:
+def gog_tags_bad() -> BeautifulSoup:
     '''Returns a bad example of a Soup of game tags.'''
     return BeautifulSoup(
         '''<span class="detail">tag7</span>

--- a/pipeline/extract/gog/conftest.py
+++ b/pipeline/extract/gog/conftest.py
@@ -56,7 +56,7 @@ def gog_description() -> BeautifulSoup:
 
 
 @pytest.fixture
-def gog_time():
+def gog_time() -> BeautifulSoup:
     '''Returns an example of release dates.'''
     return BeautifulSoup(
         '''<div class="details__content table__row-content" >

--- a/pipeline/extract/gog/conftest.py
+++ b/pipeline/extract/gog/conftest.py
@@ -25,7 +25,7 @@ def gog_fake_links_bad() -> list[BeautifulSoup]:
 
 
 @pytest.fixture
-def gog_tags() -> list[BeautifulSoup]:
+def gog_tags() -> BeautifulSoup:
     '''Returns an example of a Soup of game tags.'''
     return BeautifulSoup(
         '''<span class="details__link-text">tag7</span>

--- a/pipeline/extract/gog/conftest.py
+++ b/pipeline/extract/gog/conftest.py
@@ -1,3 +1,5 @@
+'''A conftest file that defines functions used in test_extract_gog.py.'''
+
 import pytest
 
 from bs4 import BeautifulSoup
@@ -5,6 +7,7 @@ from bs4 import BeautifulSoup
 
 @pytest.fixture
 def gog_fake_links():
+    '''Returns an example of href links to find publisher.'''
     return [BeautifulSoup(
         '<a href="/gam">seven</a>', features='html.parser').find('a'), BeautifulSoup(
         '<a href="/games?publishers=Bethesda">publisher1</a>', features='html.parser').find('a'), BeautifulSoup(
@@ -14,6 +17,7 @@ def gog_fake_links():
 
 @pytest.fixture
 def gog_fake_links_bad():
+    '''Returns badly formatted hrefs.'''
     return [BeautifulSoup(
         '<a href="/gamesubliesda">publisher1</a>', features='html.parser').find('a'), BeautifulSoup(
         '<a href="/gam">seven</a>', features='html.parser').find('a'), BeautifulSoup(
@@ -22,6 +26,7 @@ def gog_fake_links_bad():
 
 @pytest.fixture
 def gog_tags():
+    '''Returns an example of a Soup of game tags.'''
     return BeautifulSoup(
         '''<span class="details__link-text">tag7</span>
         <span class="details__link-text">tag72</span>''', features='html.parser')
@@ -29,6 +34,7 @@ def gog_tags():
 
 @pytest.fixture
 def gog_tags_bad():
+    '''Returns a bad example of a Soup of game tags.'''
     return BeautifulSoup(
         '''<span class="detail">tag7</span>
         <span class="detailt">tag72</span>''', features='html.parser')
@@ -36,6 +42,7 @@ def gog_tags_bad():
 
 @pytest.fixture
 def gog_title():
+    '''Returns an example of a game title.'''
     return BeautifulSoup(
         '''<div class="product-tile__title" title='goodtitle'>tag7</div>
         <span class="details__link-text">tag72</span>''', features='html.parser')
@@ -43,12 +50,14 @@ def gog_title():
 
 @pytest.fixture
 def gog_description():
+    '''Returns an example of a game description.'''
     return BeautifulSoup(
         '''<div class="description" >a very nice really thorough description </div>''', features='html.parser')
 
 
 @pytest.fixture
 def gog_time():
+    '''Returns an example of release dates.'''
     return BeautifulSoup(
         '''<div class="details__content table__row-content" >
         <span>{{"2024-05-24T09:55:00+03:00"}} </span>

--- a/pipeline/extract/gog/conftest.py
+++ b/pipeline/extract/gog/conftest.py
@@ -6,7 +6,7 @@ from bs4 import BeautifulSoup
 
 
 @pytest.fixture
-def gog_fake_links():
+def gog_fake_links() -> list[BeautifulSoup]:
     '''Returns an example of href links to find publisher.'''
     return [BeautifulSoup(
         '<a href="/gam">seven</a>', features='html.parser').find('a'), BeautifulSoup(

--- a/pipeline/extract/gog/conftest.py
+++ b/pipeline/extract/gog/conftest.py
@@ -25,7 +25,7 @@ def gog_fake_links_bad() -> list[BeautifulSoup]:
 
 
 @pytest.fixture
-def gog_tags():
+def gog_tags() -> list[BeautifulSoup]:
     '''Returns an example of a Soup of game tags.'''
     return BeautifulSoup(
         '''<span class="details__link-text">tag7</span>

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -14,7 +14,7 @@ def steam_rating_soup() -> BeautifulSoup:
 
 
 @pytest.fixture
-def steam_no_rating_soup():
+def steam_no_rating_soup() -> BeautifulSoup:
     '''Returns an example of html for no reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 
                         "There are no reviews for this product"

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -22,7 +22,7 @@ def steam_no_rating_soup() -> BeautifulSoup:
 
 
 @pytest.fixture
-def steam_rating_soup_positive_only():
+def steam_rating_soup_positive_only() -> BeautifulSoup:
     '''Returns an example of html for the number of reviews,
     when there are only positive reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -68,7 +68,7 @@ def steam_developer_good() -> BeautifulSoup:
 
 
 @pytest.fixture
-def steam_developer_none():
+def steam_developer_none() -> BeautifulSoup:
     '''Returns an example of html for the developer,
     when there is none.'''
     return BeautifulSoup("""<div class="review_ctn"> 

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -57,7 +57,7 @@ def steam_tags_none() -> BeautifulSoup:
 
 
 @pytest.fixture
-def steam_developer_good():
+def steam_developer_good() -> BeautifulSoup:
     '''Returns an example of html for the developer.'''
     return BeautifulSoup("""</div>
 <div class="dev_row">

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -101,7 +101,7 @@ def steam_publisher_good() -> BeautifulSoup:
 
 
 @pytest.fixture
-def search_result():
+def search_result() -> BeautifulSoup:
     '''Returns an example of html for general information about a game.'''
     return BeautifulSoup("""<span class="title">Culling of Normality</span>
 <div class="discount_final_price free">Free</div>

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -5,7 +5,7 @@ from bs4 import BeautifulSoup
 
 
 @pytest.fixture
-def steam_rating_soup():
+def steam_rating_soup() -> BeautifulSoup:
     '''Returns an example of html for the number of reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 
                          <label for="review_type_positive">Positive&nbsp;<span class="user_reviews_count">(105)</span></label><br> 

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -40,7 +40,7 @@ def steam_rating_soup_negative_only() -> BeautifulSoup:
 
 
 @pytest.fixture
-def steam_tags_good():
+def steam_tags_good() -> BeautifulSoup:
     '''Returns an example of html for game tags.'''
     return BeautifulSoup(""" <a href="https://store.steampowered.com/tags/en/Horror/?snr=1_5_9__409" class="app_tag" style="display: none;"> Horror	</a>
                          <a href="https://store.steampowered.com/tags/en/Atmospheric/?snr=1_5_9__409" class="app_tag" style="display: none;">Atmospheric</a>""",

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -48,7 +48,7 @@ def steam_tags_good() -> BeautifulSoup:
 
 
 @pytest.fixture
-def steam_tags_none():
+def steam_tags_none() -> BeautifulSoup:
     '''Returns an example of html for the tags, when there are none.'''
     return BeautifulSoup("""<div class="review_ctn"> 
                          <label for="review_type_negative">Negative&nbsp;<span class="user_reviews_count">(103)</span></label>

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -31,7 +31,7 @@ def steam_rating_soup_positive_only() -> BeautifulSoup:
 
 
 @pytest.fixture
-def steam_rating_soup_negative_only():
+def steam_rating_soup_negative_only() -> BeautifulSoup:
     '''Returns an example of html for the number of reviews,
     when there are only negative reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -6,6 +6,7 @@ from bs4 import BeautifulSoup
 
 @pytest.fixture
 def steam_rating_soup():
+    '''Returns an example of html for the number of reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 
                          <label for="review_type_positive">Positive&nbsp;<span class="user_reviews_count">(105)</span></label><br> 
                          <label for="review_type_negative">Negative&nbsp;<span class="user_reviews_count">(103)</span></label>
@@ -14,6 +15,7 @@ def steam_rating_soup():
 
 @pytest.fixture
 def steam_no_rating_soup():
+    '''Returns an example of html for no reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 
                         "There are no reviews for this product"
                          </div>""", features='html.parser')
@@ -21,6 +23,8 @@ def steam_no_rating_soup():
 
 @pytest.fixture
 def steam_rating_soup_positive_only():
+    '''Returns an example of html for the number of reviews,
+    when there are only positive reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 
                          <label for="review_type_positive">Positive&nbsp;<span class="user_reviews_count">(105)</span></label><br> 
                          </div>""", features='html.parser')
@@ -28,6 +32,8 @@ def steam_rating_soup_positive_only():
 
 @pytest.fixture
 def steam_rating_soup_negative_only():
+    '''Returns an example of html for the number of reviews,
+    when there are only negative reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 
                          <label for="review_type_negative">Negative&nbsp;<span class="user_reviews_count">(103)</span></label>
                          </div>""", features='html.parser')
@@ -35,6 +41,7 @@ def steam_rating_soup_negative_only():
 
 @pytest.fixture
 def steam_tags_good():
+    '''Returns an example of html for game tags.'''
     return BeautifulSoup(""" <a href="https://store.steampowered.com/tags/en/Horror/?snr=1_5_9__409" class="app_tag" style="display: none;"> Horror	</a>
                          <a href="https://store.steampowered.com/tags/en/Atmospheric/?snr=1_5_9__409" class="app_tag" style="display: none;">Atmospheric</a>""",
                          features='html.parser')
@@ -42,6 +49,7 @@ def steam_tags_good():
 
 @pytest.fixture
 def steam_tags_none():
+    '''Returns an example of html for the tags, when there are none.'''
     return BeautifulSoup("""<div class="review_ctn"> 
                          <label for="review_type_negative">Negative&nbsp;<span class="user_reviews_count">(103)</span></label>
                          </div>""",
@@ -50,6 +58,7 @@ def steam_tags_none():
 
 @pytest.fixture
 def steam_developer_good():
+    '''Returns an example of html for the developer.'''
     return BeautifulSoup("""</div>
 <div class="dev_row">
 <div class="subtitle column">Developer:</div>
@@ -60,6 +69,8 @@ def steam_developer_good():
 
 @pytest.fixture
 def steam_developer_none():
+    '''Returns an example of html for the developer,
+    when there is none.'''
     return BeautifulSoup("""<div class="review_ctn"> 
                          <label for="review_type_negative">Negative&nbsp;<span class="user_reviews_count">(103)</span></label>
                          </div>""",
@@ -68,6 +79,7 @@ def steam_developer_none():
 
 @pytest.fixture
 def steam_publisher_good():
+    '''Returns an example of html for the publisher.'''
     return BeautifulSoup("""[<div class="dev_row">
 <div class="subtitle column">Developer:</div>
 <div class="summary column" id="developers_list">
@@ -90,6 +102,7 @@ def steam_publisher_good():
 
 @pytest.fixture
 def search_result():
+    '''Returns an example of html for general information about a game.'''
     return BeautifulSoup("""<span class="title">Culling of Normality</span>
 <div class="discount_final_price free">Free</div>
 <div class="col search_released responsive_secondrow"> 24 Apr, 2024</div>""", features='html.parser')

--- a/pipeline/extract/steam/conftest.py
+++ b/pipeline/extract/steam/conftest.py
@@ -78,7 +78,7 @@ def steam_developer_none() -> BeautifulSoup:
 
 
 @pytest.fixture
-def steam_publisher_good():
+def steam_publisher_good() -> BeautifulSoup:
     '''Returns an example of html for the publisher.'''
     return BeautifulSoup("""[<div class="dev_row">
 <div class="subtitle column">Developer:</div>

